corehq/apps/api/resources/v0_5.py

```diff
@@ -1060,6 +1061,74 @@ def prepend_urls(self):
         ]
 
 
+@dataclass
+class NavigationEventAuditResourceParams:
+    raw_params: InitVar = None
+    users: list[str] = field(default_factory=list)
+    limit: int = None
+    local_timezone: str = None
+    cursor: str = None
+    local_date: dict[str:str] = field(default_factory=dict)
+    cursor_local_date: str = None
+    cursor_user: str = None
+
+    def __post_init__(self, raw_params=None):
+        if raw_params:
+            self._validate_keys(raw_params)
+
+            self._set_compound_keys(raw_params)
+            self.users = raw_params.getlist('users')
+            self.limit = raw_params.get('limit')
+            self.local_timezone = raw_params.get('local_timezone')
+            self.cursor = raw_params.get('cursor')
+
+    def _validate_keys(self, params):
+        valid_keys = {'users', 'limit', 'local_timezone', 'cursor', 'format', 'local_date'}
+        standardized_keys = set()
+
+        for key in params.keys():
+            if '.' in key:
+                key, qualifier = key.split('.', maxsplit=1)
+            standardized_keys.add(key)
+
+        invalid_keys = standardized_keys - valid_keys
+        if invalid_keys:
+            raise ValueError(f"Invalid parameter(s): {', '.join(invalid_keys)}")
+
+    def _set_compound_keys(self, params):
+        local_date = {}
+        for key in params.keys():
+            if '.' in key:
+                prefix, qualifier = key.split('.', maxsplit=1)
+                if prefix == 'local_date':
+                    local_date[key] = params.get(key)
```

Comment:
so this stores a field like this, right?
```python
    local_date = {
        'local_date.gte': '2023-01-01',
    }
```
why not use `qualifier` as the dict key, since you already have it and it's already namespaced?  Then you could avoid splitting it back apart later when converting it to a filter.  (unless that makes it easier to reuse helper functions or something)

Reply:
Good call, thanks 
[5f978c5](https://github.com/dimagi/commcare-hq/pull/33077/commits/5f978c51e0b955736589a46f2e83afeb5838411d)

Reply:
Good call, thanks 
[5f978c5](https://github.com/dimagi/commcare-hq/pull/33077/commits/5f978c51e0b955736589a46f2e83afeb5838411d)